### PR TITLE
add start_timestep signal

### LINKED ISSUE #5078: connect edit_parameters test to different signal
Following #5070 the test `edit_parameters` no longer does what it did before, because it was connected to a signal that is no longer triggered. We need to fix this by connecting the test to a different signal that triggers every timestep.

### PR DESCRIPTION
**Pull request by tjhei**
fixes #5078

### CODE REVIEW

**Review by gassmoeller — APPROVED** [review]
Makes sense, just a question about the include.

File: tests/edit_parameters.cc [hunk 1]
```diff
@@ -18,6 +18,7 @@
   <http://www.gnu.org/licenses/>.
 */
 
+#include "aspect/parameters.h"
```
[the remark this answers is not in the document]
Comment (tjhei) [reply tests/edit_parameters.cc:21]: ouch, trying out VS code and it decided to add this (and I didn't notice).

**Comment by tjhei** [comment]
I am not too happy with this test, because I dislike the const_cast but I also did not want to expose a writable ``Parameters`` object to the signal. There is no guarantee that any of the edits done to the parameters object do anything (in fact most edits are probably going to break stuff).

**Review by gassmoeller — APPROVED** [review]
I think the current form is fine. It allows read access to everything necessary, and like you say, if users want to change things they need to know what they are doing because it likely breaks stuff. So hiding it behind a const_cast is a good thing (it will be rarely used anyway).

**Comment by tjhei** [comment]
Ok, great. I addressed your comments.

**Pull request merged**